tests/components/dexcom/test_sensor.py

```diff
@@ -58,36 +53,3 @@ async def test_sensors_update_failed(hass: HomeAssistant) -> None:
     assert test_username_glucose_value.state == STATE_UNAVAILABLE
     test_username_glucose_trend = hass.states.get("sensor.test_username_glucose_trend")
     assert test_username_glucose_trend.state == STATE_UNAVAILABLE
-
-
-async def test_sensors_options_changed(hass: HomeAssistant) -> None:
-    """Test we handle sensor unavailable."""
-    entry = await init_integration(hass)
```

Comment:
I wasn't sure how to add a test to ensure that the previously selected unit was kept after this update.

I had a quick attempt with the code below, but I couldn't get it to work :(

```
entity_registry.async_get_or_create(
        domain=DOMAIN,
        platform=Platform.SENSOR,
        unique_id="test_username_glucose_value",
        unit_of_measurement=original_unit,
    )
```

Reply:
If both values are valid for the device class it might just reset to the native unit, but I think the history should be fine

Reply:
I'd love it if @emontnemery could take a look and confirm, as I worked on this before with him but I just don't remember.

Reply:
This will work 👍 
We'll handle the conversion needed.